tests/integration/conftest.py

```diff
@@ -152,3 +152,34 @@ async def _wait_for_ingress(ingress_name):
         )
 
     return _wait_for_ingress
+
+
+@fixture(scope="module")
+def get_ingress_annotation(ops_test: OpsTest):
+    """Create a function that will retrieve all annotation from a ingress by its name."""
+    assert ops_test.model
+    kubernetes.config.load_kube_config()
+    kube = kubernetes.client.NetworkingV1Api()
+    model_name = ops_test.model.name
+
+    def _get_ingress_annotation(ingress_name: str):
+        return kube.read_namespaced_ingress(
+            ingress_name, namespace=model_name
+        ).metadata.annotations
+
+    return _get_ingress_annotation
```

Comment:
I'm wondering if these would be better served as helper functions rather than as fixture?
Would there be a standard or guide on these? I think having them as helper functions can remove scoping and dependencies on tests. Also it could be easier to navigate code through IDE.

Reply:
The problem of converting this as a helper function is that this function needs the `ops_test` fixture. 
If I am going to declare this as a helper function, I have to pass the `ops_test` every time I invoke this function.
While as a fixture, I can just use the `ops_test` fixture.
Both way will do, it's just a personal preference.
For the IDE part, I think there are plugins for pytest that can do fixture navigation, my IDE can do that by default.